src/simulator/ackermann_model.cpp

```diff
@@ -0,0 +1,96 @@
+#include "simulator/ackermann_model.h"
+#include "config_reader/config_reader.h"
+#include "shared/util/timer.h"
+#include "shared/math/math_util.h"
+
+using std::string;
+using f1tenth_simulator::AckermannCurvatureDriveMsg;
+using math_util::AngleDiff;
+using math_util::AngleMod;
+
+namespace ackermann {
+
+CONFIG_FLOAT(cMinTurnR, "min_turn_radius");
+CONFIG_FLOAT(cMaxAccel, "max_accel");
+CONFIG_FLOAT(cMaxSpeed, "max_speed");
+CONFIG_FLOAT(cAngularErrorBias, "angular_error_bias");
+CONFIG_FLOAT(cAngularErrorRate, "angular_error_rate");
+CONFIG_STRING(cDriveTopic, "drive_callback_topic");
+
+
+AckermannModel::AckermannModel(const string& config_file, ros::NodeHandle n) :
+    angular_error_(0, 1) {
+  config_reader::ConfigReader({config_file});
+  // Use the config reader to initialize the subscriber
+  drive_subscriber_ = n.subscribe(
+      cDriveTopic,
+      1,
+      &AckermannModel::DriveCallback,
+      this);
+  template_lines_ = {};
+  pose_lines_ = {};
+}
```

Comment:
I think this is basically how we want to follow for all other robots to load configs? I'll follow this style unless Joydeep has other comments on this.

Reply:
I think so yes. Barring some issue I haven't thought of this seems to be how all motion models should load their configs and topics.